kaldilm/csrc/arpa_lm_compiler.cc

```diff
@@ -118,13 +118,33 @@ class OptimizedHistKey {
 
 }  // namespace
 
+class ArpaLmCompilerImplInterface {
+ public:
+  virtual ~ArpaLmCompilerImplInterface() = default;
```

Comment:
any chance of some short comments here explaining the interface?
the only other comment I have is, of course check it doesn't change the behavior for normal inputs.